### ISSUE TITLE
🗺 Streaming/Defer

### ISSUE THREAD

**Issue by ryanflorence** [remix-run/remix#4486]
Adding `defer`, `<Await>`, etc. from React Router 6.4 and React 18 streaming support for document requests to Remix.

Depends on #4485 

## Progress

- [x] RFC n/a
- [x] PR
    - https://github.com/remix-run/react-router/pull/9760
    - https://github.com/remix-run/remix/pull/4920
- [x] Prerelease
- [x] Move existing docs - @jacob-ebey 
- [ ] Needs new docs written - @ryanflorence 

## Issues/PRs

**Comment by ryanflorence** [comment]
@brophdawg11 to get with @jacob-ebey to work out the React Router release we need for this work, and moving over deferred branch test suite and open issues.

**Comment by jacob-ebey** [comment]
This work is included in experimental release: 0.0.0-experimental-2f4891673

**Comment by jacob-ebey** [comment]
@ryanflorence we need to bring the docs over. I have a fair amount I can link you to if you want to wrangle them up into the best places across remix and react router.

**Comment by jacob-ebey** [comment]
This has been released in `1.11.0-pre.0`

**Comment by jacob-ebey** [comment]
@ryanflorence I'll let you close this after you take a pass through the docs.

**Comment by ryanflorence** [comment]
made a note in my tasks, so gonna close this out, we have *some* docs (thanks!) so let's call this done!
